test/box-luatest/gh_9406_fix_use_after_free_on_shutdown_test.lua

```diff
@@ -0,0 +1,24 @@
+local t = require('luatest')
+
+local g = t.group()
+
+g.test_shutdown_with_lingering_fiber_join = function()
+    local script = [[
+        local fiber = require('fiber')
+
+        local f = nil
+        fiber.create(function()
+            fiber.sleep(0.1)
+            fiber.join(f)
+        end)
+        f = fiber.create(function()
+            fiber.sleep(1000)
+        end)
+        f:set_joinable(true)
+        fiber.sleep(0.2)
+        os.exit()
+    ]]
+    local tarantool_bin = arg[-1]
+    local cmd = string.format('%s -e "%s"', tarantool_bin, script)
+    t.assert(os.execute(cmd) == 0)
```

Comment:
Why no changelog? AFAICS this may result in a crash.

Reply:
The problem code is executed on finish steps of shutdown when we are only freeing memory. So I guess so we will not overwrite some allocated memory. But let's add changelog anyway.

```md
## bugfix/core

* Fixed potential use-after-free on Tarantool shutdown with lingering
  fiber join (gh-9406).
```